salt/orchestrate/aws/mitx.sls

```diff
@@ -0,0 +1,369 @@
+# Make sure that instance profiles exist for node types so that
+# they can be granted access via permissions attached to those
+# profiles because it's easier than managing IAM keys
+{% from "orchestrate/aws_env_macro.jinja" import VPC_NAME, VPC_RESOURCE_SUFFIX,
+ ENVIRONMENT, subnet_ids with context %}
+
+{% for profile in ['consul', 'mongodb', 'rabbitmq', 'edx'] %}
+ensure_instance_profile_exists_for_{{ profile }}:
+  boto_iam_role.present:
+    - name: {{ profile }}-instance-role
+{% endfor %}
+
+{% if VPC_RESOURCE_SUFFIX == 'mitx-qa' %}
+{% set cidr_block_public_subnet_1 = 10.5.1.0/24 %}
+{% set cidr_block_public_subnet_2 = 10.5.2.0/24 %}
+{% set cidr_block_public_subnet_3 = 10.5.3.0/24 %}
+{% set cidr_ip = 10.5.0.0/22 %}
+{% set cidr_block = 10.5.0.0/16 %}
+{% elif VPC_RESOURCE_SUFFIX == 'mitx-rp' %}
+{% set cidr_block_public_subnet_1 = 10.6.1.0/24 %}
+{% set cidr_block_public_subnet_2 = 10.6.2.0/24 %}
+{% set cidr_block_public_subnet_3 = 10.6.3.0/24 %}
+{% set cidr_ip = 10.6.0.0/22 %}
+{% set cidr_block = 10.5.0.0/16 %}
+{% endif %}
```

Comment:
Rather than using this mapping you can use the network prefix values stored in the `environments` pillar data.